### PR TITLE
Add f80

### PR DESCRIPTION
**Pull request by Vexu**
As discussed in https://github.com/Vexu/arocc/issues/213#issuecomment-1016037714

Should calls to compiler-rt on vectors be generated as unrolled or regular loops?

### CODE REVIEW

**Review by andrewrk — DISMISSED** [review]
The changes look great :+1: 

> Should calls to compiler-rt on vectors be generated as unrolled or regular loops?

I suggest unrolled, since vectors lengths are generally small numbers. I think the lang spec will allow an implementation to do either way and still be correct.

**Comment by andrewrk** [comment]
Some behavior tests to go along with this would be lovely.

File: doc/langref.html.in [hunk 1]
```diff
@@ -737,6 +737,11 @@ pub fn main() void {
           <td><code class="c">double</code></td>
           <td>64-bit floating point (52-bit mantissa) IEEE-754-2008 binary64</td>
         </tr>
+        <tr>
+            <th scope="row">{#syntax#}f80{#endsyntax#}</th>
+          <td><code class="c">double</code></td>
+          <td>64-bit floating point (52-bit mantissa) IEEE-754-2008 80-bit extended precision</td>
```
Comment (ehaas) [inline doc/langref.html.in:743]: The bit sizes seem like a copy-paste error?

**Comment by andrewrk** [comment]
Looks like the branch is tripping an LLVM assert. This can be reproduced by building against a debug build of LLVM:

```
#4  0x000000000801f9e1 in llvm::ConstantExpr::getBitCast (C=0x35a8b170, DstTy=0xeef8268, 
    OnlyIfReduced=false) at /home/andy/Downloads/llvm-project-13/llvm/lib/IR/Constants.cpp:2225
2225      assert(CastInst::castIsValid(Instruction::BitCast, C, DstTy) &&
(gdb) 
#5  0x00000000080467e4 in LLVMConstBitCast (ConstantVal=0x35a8b170, ToType=0xeef8268)
    at /home/andy/Downloads/llvm-project-13/llvm/lib/IR/Core.cpp:1766
1766      return wrap(ConstantExpr::getBitCast(unwrap<Constant>(ConstantVal),
(gdb) 
#6  0x0000000001a4be98 in gen_const_val (g=0xeeeeea0, const_val=0x2c1ee6d0, name=0x2c19e270 "nan_f80")
    at /home/andy/Downloads/zig/src/stage1/codegen.cpp:8100
8100                        return LLVMConstBitCast(as_int, get_llvm_type(g, type_entry));
(gdb) 
#7  0x0000000001a4dede in render_const_val (g=0xeeeeea0, const_val=0x2c1ee6d0, 
    name=0x2c19e270 "nan_f80") at /home/andy/Downloads/zig/src/stage1/codegen.cpp:8560
8560            const_val->llvm_value = gen_const_val(g, const_val, name);
(gdb) 
#8  0x0000000001a4ee03 in do_code_gen (g=0xeeeeea0)
    at /home/andy/Downloads/zig/src/stage1/codegen.cpp:8770
8770                render_const_val(g, var->const_value, symbol_name);
(gdb) p symbol_name
$1 = 0x2c19e270 "nan_f80"
(gdb) 
```

**Review by andrewrk — APPROVED** [review]

**Pull request merged**